control/src/Controller/CommonSMC.cpp

```diff
@@ -156,7 +156,6 @@ void CommonSMC::initialize(const Model& m)
   }
 
   _SMC->nonSmoothDynamicalSystem()->insertDynamicalSystem(_DS_SMC);
-  _SMC->nonSmoothDynamicalSystem()->topology()->setOSI(_DS_SMC, _integratorSMC);
```

Comment:
Hm, in this case I skipped it because in fact it is too early for initializeOSIforDS to call initializeDynamicalSystem, which is okay to defer to Model/Simulation::initialize(), but I forgot that this would skip the OSI association. In fact Simulation::initialize() assigns the first OSI if it is not done prior.  It would be nice to do the same thing in *all cases* from the user code point of view, so perhaps it can call initializeOSIforDS() in any case, and it will simply skip initializeDynamicalSystem if the OSI is not yet initialized.  Then initializeDynamicalSystem will be called by OSI::initialize() during Simulation::initialize().